demos/TicTacToe/Classes/MainMenuScene.h

```diff
@@ -2,18 +2,60 @@
 #define TICTACTOE_DEMO_CLASSES_MAINMENU_SCENE_H_
 
 #include "cocos2d.h"
+#include "firebase/auth.h"
+#include "firebase/database.h"
+#include "firebase/future.h"
+#include "firebase/util.h"
 #include "ui/CocosGUI.h"
+using std::to_string;
+
+void LogMessage(const char*, ...);
+bool ProcessEvents(int);
+void WaitForCompletion(const firebase::FutureBase&, const char*);
+
+std::string GenerateUid(std::size_t);
 
 class MainMenuScene : public cocos2d::Layer, public cocos2d::TextFieldDelegate {
  public:
   // Builds a simple scene that uses the bottom left cordinate point as (0,0)
   // and can have sprites, labels and nodes added onto it.
   static cocos2d::Scene* createScene();
+  virtual void MainMenuScene::update(float);
+  virtual void MainMenuScene::onEnter();
+  void MainMenuScene::updateUserRecord();
+  void MainMenuScene::initializeUserRecord();
   // Initializes the instance of a Node and returns a boolean based on if it was
   // successful in doing so.
   virtual bool init();
   // Defines a create type for a specific type, in this case a Layer.
   CREATE_FUNC(MainMenuScene);
+  cocos2d::DrawNode* auth_background;
+  cocos2d::Node auth_node;
+  cocos2d::Label* invalid_login_label;
+  cocos2d::Label* user_record_label;
+  cocos2d::TextFieldTTF* email_text_field;
+  cocos2d::TextFieldTTF* password_text_field;
+  // Defines the stage that the class is in.
+  enum stageEnum {
```

Comment:
usually enum definitions and delcarations have their own subsection devoted to them. Here its in the middle of a bunch of variable declarations. 

I recommend make a section of:
enums
{new line separator}
methods
{new line separator}
variables
{new line separator}

Reply:
note that this would repeat based on public and then private definitions:

public:
 {as above, enums, methods, variables}

private:
  {enums, metods, variables}